### PR TITLE
Add veil to filters if no incidents

### PR DESCRIPTION
**Pull request by jgoley**
[PT Bug](https://www.pivotaltracker.com/story/show/149962130)

### CODE REVIEW

**Comment by nathanathan** [comment]
This makes it so there is no way to change the filters if a filter is selected that removes all the incidents.

**Comment by jgoley** [comment]
My bad. The latest commit fixes that by using the `hasNoIncidents` method on the instance. It also adds the same logic to the smart event page.

File: client/controllers/smartEvents/smartEvent.coffee [hunk 1]
```diff
@@ -11,6 +12,10 @@ Template.smartEvent.onCreated ->
   @filterQuery = new ReactiveVar({})
   @selectedIncidentTypes = new ReactiveVar([])
 
+  @hasNoIncidents = =>
+    filterQuery = @filterQuery.get()
+    _.isEmpty(filterQuery) and not EventIncidents.find(filterQuery).count()
```
Comment (nathanathan) [inline client/controllers/smartEvents/smartEvent.coffee]: Does the filterQuery being non-empty imply that the event has incidents?

Comment (jgoley) [inline client/controllers/smartEvents/smartEvent.coffee]: Oh, right. I guess we only need to check if the empty query has a count.
This brings up an issue—if the query is empty (the user creates a smart event with only a title) doesn't the smart event page try to subscribe to every incident?

**Comment by nathanathan** [comment]
👍

**Pull request merged**